llama-index-core/llama_index/core/workflow/workflow.py

```diff
@@ -28,13 +28,22 @@ def __init__(
         self._step_flags: Dict[str, asyncio.Event] = {}
         self._accepted_events: List[Tuple[str, str]] = []
         self._retval: Any = None
+        self._step_functions: Dict[str, Callable] = {}
+
+    def add_step(self, func: Callable) -> None:
+        """Adds a free function as step for this workflow instance."""
+        self._step_functions[func.__name__] = func
+
+    def _get_steps(self) -> Dict[str, Callable]:
+        """Returns all the steps, whether defined as methods or free functions."""
+        return {**get_steps_from_class(self), **self._step_functions}
```

Comment:
probably the step names from both sources could collide, but maybe thats an edge case for now (that's actually why I had to change `def done()` to `def _done()` earlier)

Reply:
Good catch, agree it's an edge case but let's at least raise an error for now